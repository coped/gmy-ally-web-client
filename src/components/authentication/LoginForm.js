import React, { Component } from "react";
import TextInputField from "components/common/TextInputField";
import Button from "components/common/Button";
import Tag from "components/common/Tag";
import AsyncRequest from "lib/asyncRequest";
import { endpoints } from "lib/endpoints";
import Messages from "lib/messages";

export default class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      emailValue: "",
      passwordValue: "",
      apiMessages: "",
    };

    this.onTextFieldChange = this.onTextFieldChange.bind(this);
    this.loginUser = this.loginUser.bind(this);
  }

  onTextFieldChange(event) {
    const target = event.target;
    this.setState({
      [target.name]: target.value,
    });
  }

  loginUser(credentials) {
    AsyncRequest.post(endpoints.authentication.login, credentials)
      .then((response) => response.json())
      .then((data) => {
        if (data.status === "success") {
          this.props.isAuthenticated(data.payload.jwt);
        } else {
          this.setState({ apiMessages: data.messages });
        }
      })
      .catch((error) => {
        console.log(error);
        this.setState({ apiMessages: [Messages.generalError] });
      });
  }

  render() {
    const { toggleSignupForm } = this.props;
    const { emailValue, passwordValue, apiMessages } = this.state;
    return (
      <div>
        <form
          onSubmit={(e) => {
            this.loginUser({
              login: {
                email: emailValue,
                password: passwordValue,
              },
            });
            e.preventDefault();
          }}
        >
          <div className="has-text-centered">
            {apiMessages &&
              apiMessages.map((message, index) => (
                <Tag
                  key={index}
                  type={message.type}
                  message={message.message}
                />
              ))}
          </div>
          <TextInputField
            label="Email:"
            name="emailValue"
            type="email"
            placeholder="your@email.com"
            value={emailValue}
            onChange={this.onTextFieldChange}
          />
          <TextInputField
            label="Password:"
            name="passwordValue"
            type="password"
            value={passwordValue}
            onChange={this.onTextFieldChange}
          />
          <div className="field">
            <div className="control">
              <Button textContent="Log in" classModifiers="is-link" />
            </div>
          </div>
          <div>
            <p className="is-size-6">
              Don't have an account? <a onClick={toggleSignupForm}>Sign up.</a>
            </p>
          </div>
        </form>
      </div>
    );
  }
}
