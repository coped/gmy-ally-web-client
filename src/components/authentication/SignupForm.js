import React, { Component } from 'react';
import TextInputField from 'components/common/TextInputField';
import Button from 'components/common/Button';

export default class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
    };
  }

  render() {
    return (
      <div className="container">
        <TextInputField 
          label="Name:"
          type="text"
          placeholder="Your Name"
        />
        <TextInputField
          label="Email:"
          type="email"
          placeholder="your@email.com"
        />
        <TextInputField
          label="Password:"
          type="password"
        />
        <TextInputField
          label="Password confirmation"
          type="password"
        />
        <div className="field">
          <div className="control">
            <Button 
              textContent="Log in"
              classModifiers="is-link"
              onClick={() => console.log('clicked')}
            />
          </div>
        </div>
        <div>
          <p className="is-size-6">Already have an account? <a onClick={this.props.toggleSignupCallback}> 
            Log in.</a>
          </p>
        </div>
      </div>
    );
  }
}
