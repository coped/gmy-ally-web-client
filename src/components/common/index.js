import Notification from "components/common/Notification";
import Button from "components/common/Button";
import PageLoader from "components/common/PageLoader";

export { Notification, Button, PageLoader };
