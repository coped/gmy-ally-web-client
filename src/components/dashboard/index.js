import Dashboard from "components/dashboard/Dashboard";

export { Dashboard };
