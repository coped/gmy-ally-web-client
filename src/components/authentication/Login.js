import React, { Component } from 'react';
import Logo from 'components/common/Logo';
import LoginForm from './LoginForm';
import SignupForm from './SignupForm';

export default class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showSignup: false
    };

    this.toggleSignupForm = this.toggleSignupForm.bind(this);
  }

  toggleSignupForm() {
    this.setState({ showSignup: !this.state.showSignup});
  }

  render() {
    const {showSignup} = this.state;
    return (
      <div className="container">
        <div className="logo-medium">
          <Logo
            isAnimated={true}
          />
        </div>
        <div className="container box">
          {(showSignup) ? 
          <SignupForm 
            toggleSignupCallback={this.toggleSignupForm}
          /> : 
          <LoginForm 
            toggleSignupCallback={this.toggleSignupForm}
          />}
        </div>
      </div>
    );
  }
}
