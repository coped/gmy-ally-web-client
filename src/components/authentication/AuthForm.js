import React, { Component } from "react";
import { Logo } from "components/common/";
import { Login, Signup } from "components/authentication";

export default class AuthForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showSignup: false,
    };

    this.toggleSignupForm = this.toggleSignupForm.bind(this);
    this.showForm = this.showForm.bind(this);
  }

  showForm() {
    if (this.state.showSignup) {
      return (
        <Signup
          toggleSignupForm={this.toggleSignupForm}
          authenticate={this.props.authenticate}
        />
      );
    } else {
      return (
        <Login
          toggleSignupForm={this.toggleSignupForm}
          authenticate={this.props.authenticate}
        />
      );
    }
  }

  toggleSignupForm() {
    this.setState({ showSignup: !this.state.showSignup });
  }

  render() {
    return (
      <div className="center-column">
        <div className="logo-login">
          <Logo isAnimated={true} />
        </div>
        <div className="column is-5">
          <div className="box">{this.showForm()}</div>
        </div>
      </div>
    );
  }
}
