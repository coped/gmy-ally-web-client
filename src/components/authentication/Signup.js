import React, { Component } from "react";
import { TextInputField, Button } from "components/form";
import { AsyncRequest, endpoints, Messages } from "lib";
import { Notification } from "components/common";

export default class Signup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      nameValue: "",
      emailValue: "",
      passwordValue: "",
      passwordConfirmationValue: "",
      apiMessages: "",
      clientMessages: "",
    };

    this.onTextFieldChange = this.onTextFieldChange.bind(this);
    this.signupUser = this.signupUser.bind(this);
  }

  signupUser(credentials) {
    this.setState({ isLoading: true });
    AsyncRequest.post(endpoints.users.create, credentials)
      .then((response) => response.json())
      .then((data) => {
        if (data.status === "success") {
          this.props.authenticate(data);
        } else {
          this.setState({ apiMessages: data.messages });
        }
      })
      .catch((error) => {
        console.log(error);
        this.setState({ clientMessages: [Messages.connectionError] });
      })
      .finally(() => this.setState({ isLoading: false }));
  }

  onTextFieldChange(event) {
    const target = event.target;
    this.setState({
      [target.name]: target.value,
    });
  }

  render() {
    const { toggleSignupForm } = this.props;
    const {
      nameValue,
      emailValue,
      passwordValue,
      passwordConfirmationValue,
      apiMessages,
      clientMessages,
    } = this.state;
    const isLoading = this.state.isLoading ? "is-loading" : "";
    return (
      <div>
        <form
          onSubmit={(e) => {
            this.signupUser({
              user: {
                name: nameValue,
                email: emailValue,
                password: passwordValue,
                password_confirmation: passwordConfirmationValue,
              },
            });
            e.preventDefault();
          }}
        >
          <div>
            {apiMessages &&
              apiMessages.map((notification, index) => (
                <Notification type={notification.type}>
                  <p>The following errors occurred:</p>
                  <ul>
                    {notification.message.split("\n").map((error) => (
                      <li>â€¢ {error}</li>
                    ))}
                  </ul>
                </Notification>
              ))}
            <div className="has-text-centered">
              {clientMessages &&
                clientMessages.map((notification, index) => (
                  <Notification type={notification.type}>
                    <p>{notification.message}</p>
                  </Notification>
                ))}
            </div>
          </div>
          <TextInputField
            label="Name:"
            name="nameValue"
            type="text"
            placeholder="Your Name"
            value={nameValue}
            onChange={this.onTextFieldChange}
          />
          <TextInputField
            label="Email:"
            name="emailValue"
            type="email"
            placeholder="your@email.com"
            value={emailValue}
            onChange={this.onTextFieldChange}
          />
          <TextInputField
            label="Password:"
            name="passwordValue"
            type="password"
            value={passwordValue}
            onChange={this.onTextFieldChange}
          />
          <TextInputField
            label="Password confirmation:"
            name="passwordConfirmationValue"
            type="password"
            value={passwordConfirmationValue}
            onChange={this.onTextFieldChange}
          />
          <Button isLoading={isLoading} classModifiers={"is-link"}>
            Sign up
          </Button>
          <div>
            <p className="is-size-6">
              Already have an account? <a onClick={toggleSignupForm}>Log in.</a>
            </p>
          </div>
        </form>
      </div>
    );
  }
}