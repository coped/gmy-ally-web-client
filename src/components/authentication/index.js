import AuthForm from "components/authentication/AuthForm";
import Login from "components/authentication/Login";
import Signup from "components/authentication/Signup";

export { AuthForm, Login, Signup };
