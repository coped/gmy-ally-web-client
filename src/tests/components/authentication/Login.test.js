import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Login } from "components/authentication";

Enzyme.configure({ adapter: new Adapter() });

describe("Login", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Login />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders a form", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.find("form").length).toEqual(1);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<Login />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
