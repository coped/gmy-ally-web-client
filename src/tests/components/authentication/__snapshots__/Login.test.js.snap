// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login has a valid snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="has-text-centered"
    >
      
    </div>
    <div
      className="field"
    >
      <label
        className="label"
      >
        Email:
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          name="emailValue"
          onChange={[Function]}
          placeholder="your@email.com"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <label
        className="label"
      >
        Password:
      </label>
      <div
        className="control"
      >
        <input
          className="input"
          name="passwordValue"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <button
          className="button is-link"
        >
          Log in
        </button>
      </div>
    </div>
    <div>
      <p
        className="is-size-6"
      >
        Don't have an account?
         
        <a
          id="show-signup"
        >
          Sign up.
        </a>
      </p>
    </div>
  </form>
</div>
`;
